src/Codeception/Module/Symfony2.php

```diff
@@ -174,6 +178,46 @@ protected function getKernelClass()
     }
 
     /**
+     * Get router from container.
+     *
+     * @return object
+     */
+    private function getNonCachedRouter()
+    {
+        if (!$this->kernel->getContainer()->has('router')) {
+            $this->fail('Router not found.');
+        }
+
+        return $this->kernel->getContainer()->get('router');
+    }
+
+    /**
+     * Helper method to get router.
+     * Router can be cached or not by configuration.
+     *
+     * @return object
+     */
+    protected function getRouter()
+    {
+        if ($this->config['cache_router'] === true) {
+            if (null === self::$router) {
+                self::$router = $this->getNonCachedRouter();
+            }
+            return self::$router;
+        }
+
+        return $this->getNonCachedRouter();
+    }
+
+    /**
+     * Invalidate previously cached router.
+     */
+    public function invalidateCachedRouter()
```

Comment:
I'd rather not make it public. This method is useless if you have disabled cached routes